gcc/rust/rust-session-manager.h

```diff
@@ -205,9 +208,31 @@ struct CompileOptions
     enable_dump_option (DumpOption::TYPE_RESOLUTION_DUMP);
   }
 
+  /* Validate the crate name using the ASCII rules
+   TODO: Support Unicode version of the rules */
+  bool validate_crate_name (const std::string crate_name)
+  {
+    if (crate_name.length () > kMaxNameLength || crate_name.empty ())
+      return false;
+    for (auto &c : crate_name)
+      {
+	if (!(ISALNUM (c) || c == '_' || c == '-'))
+	  return false;
```

Comment:
I think it would be good to emit the error here rather than when checking for the return value of `set_crate_name`? This way, we could have errors saying ("crate name cannot be empty", "crate name cannot be longer than 64 characters", "invalid character in crate name: @") and so on

Reply:
You are right! I will change the implementation to emit an error here.